usaspending_api/transactions/management/commands/delete_procurement_records.py

```diff
@@ -20,51 +18,31 @@ class Command(AgnosticDeletes, BaseCommand):
     destination_table_name = SourceProcurementTransaction().table_name
     shared_pk = "detached_award_procurement_id"
 
-    def fetch_deleted_transactions(self, date):
+    def fetch_deleted_transactions(self, date_time: datetime) -> Optional[dict]:
         ids_to_delete = defaultdict(list)
-        regex_str = ".*_delete_records_(IDV|award).*"
 
         if settings.IS_LOCAL:
             logger.info("Local mode does not handle deleted records")
             return None
 
-        s3 = boto3.resource("s3", region_name=settings.USASPENDING_AWS_REGION)
-        s3_bucket = s3.Bucket(settings.FPDS_BUCKET_NAME)
+        full_key_list = retrieve_deleted_fpds_transactions(start_datetime=date_time)
 
-        # Only use files that match the date we're currently checking
-        for obj in s3_bucket.objects.all():
-            key_name = obj.key
-            if "/" in key_name or not re.search(regex_str, key_name):
-                continue
-
-            file_date = datetime.strptime(key_name[: key_name.find("_")], "%m-%d-%Y").date()
-
-            if file_date < date:
-                continue
-
-            logger.info("========= {}".format(key_name))
-            data = io.BytesIO()
-            s3_bucket.download_fileobj(key_name, data)
-            data.seek(0)
-            reader = csv.reader(data.read().decode("utf-8").splitlines())
-            next(reader)  # skip the header
-
-            full_key_list = [rows[0] for rows in reader]
-            numeric_keys = list([int(rows) for rows in full_key_list if rows.isnumeric()])
-            odd_ids = set(full_key_list).symmetric_difference([str(id) for id in numeric_keys])
+        for date, transaction_ids in full_key_list.items():
+            numeric_keys = list([int(row) for row in transaction_ids if row.isnumeric()])
+            odd_ids = set(transaction_ids).symmetric_difference([str(id) for id in numeric_keys])
```

Comment:
Is the objective of this just to get the non-numeric keys? If not, this code's purpose is unclear.

Reply:
```python
    if odd_ids:
                logger.info(f"Unexpected non-numeric IDs in file: {list(odd_ids)}")
```

Reply:
the couldn't odd_ids just be `list([int(row) for row in transaction_ids if not row.isnumeric()])`? That would be clearer IMO.

Reply:
I don't recall the original reasoning for the method. It is possible the original justification no longer holds. This PR is just altering the variable name in those lines.